### ISSUE TITLE
Is nanoTabICL restricted to fixed number of classes?

### ISSUE THREAD

**Issue by liam-sbhoo** [soda-inria/nanotabicl#1]
Hi there!

Thanks for making nanoTabICL!

It seems that nanoTabICL only supports a fixed number of classes only (e.g. if the model is configured to 10 classes, then it can only do inference on classification dataset with 10 classes)? Or am I missing something?

Thanks!

**Comment by dholzmueller** [comment]
Hi Liam, you can use a number of classes K that is less than the max number of classes by simply passing in a vector `y_train` that only has values in {0 ..., K-1}, and restricting the output logits to the first K dimensions.

**Comment by liam-sbhoo** [comment]
I see, got it. Thanks!
